src/proposals/Proposals.jl

```diff
@@ -248,5 +249,149 @@ function (prop::RStagger)(rng::AbstractRNG,
     return u, v, logl, ncall
 end
       
+"""
+    Proposals.Slice(;slices=5, scale=1)
+
+Propose a new live point by a series of random slices away from an existing live point.
+This is a standard _Gibbs-like_ implementation where a single multivariate slice is a combination of `slices` univariate slices through each axis.
+
+## Parameters
+- `slices` is the minimum number of slices
+- `scale` is the proposal distribution scale, which will update _between_ proposals.
+"""
+@with_kw mutable struct Slice <: AbstractProposal
+    slices = 5
+    scale = 1.0
+    
+    @assert slices ≥ 1 "Number of slices must be greater than or equal to 1"
+    @assert scale ≥ 0 "Proposal scale must be non-negative"
+end
 
+function (prop::Slice)(rng::AbstractRNG,
+                       point::AbstractVector,
+                       logl_star,
+                       bounds::AbstractBoundingSpace,
+                       loglike,
+                       prior_transform;
+                       kwargs...)
+    # setup
+    n = length(point)
+    nc = nexpand = ncontract = 0
+    fscale = [] 
+    axlens = []
+    local idxs, r, u, u_prop, v_prop, logl_prop, logl_l, logl_r
+    
+    # modifying axes and computing lengths
+    axes = prop.scale .* Bounds.tran_axes(bounds)'
+    axlens = [norm(@view(axes[i, :])) for i in Base.axes(axes, 1)]
+    
+    # slice sampling loop
+    for it in 1:prop.slices
+        
+        # shuffle axis update order
+        idxs = shuffle!(rng, Vector(1:n))
```

Comment:
switch to using an index agnostic system
```suggestion
        idxs = shuffle!(rng, Base.axes(axes, 1))
```